k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -55,6 +56,21 @@ class EndpointsNamer(
       Activity.value(NameTree.Neg)
   }
 
+  override def getAllNames: Activity[Set[Path]] = {
```

Comment:
I think this can be a (lazy?) val.  Since vars/activities are populated lazily, it's safe to build statically.  Furthermore, this ensures that multiple consumers share observations, potentially reducing redundant work.